website/content/docs/k8s/operations/gossip-encryption-key-rotation.mdx

```diff
@@ -10,111 +10,133 @@ The following instructions provides a step-by-step manual process for rotating [
 
-  ```shell-session
-  consul keyring -list
-  ==> Gathering installed encryption keys...
+        ```shell-session
+        export CONSUL_HTTP_TOKEN=<INSERT BOOTSTRAP ACL TOKEN>
+        ```
+    
+    1. Install the new Gossip encryption key with the `consul keyring` command:
+
+        ```shell-session
+        consul keyring -install="Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w="
+        ==> Installing new gossip encryption key...
+        ```
+        Consul automatically propagates this encryption key across all Clients and Servers across the cluster and the federation if Consul federation is enabled.
+
+    1. List the keys in the keyring to verify the new key has been installed successfully.
+
+        ```shell-session
+        consul keyring -list
+        ==> Gathering installed encryption keys...
+
+        WAN:
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [2/2]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [2/2]
+
+        dc1 (LAN):
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [4/4]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [4/4]
+
+        dc2 (LAN):
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [4/4]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [4/4]
+        ```
+
+1. Use the new key as the gossip encryption key.
+
+    1. Once the new key has been added to the keychain, you can install it as the new gossip encryption key. In order to do so
+    run the following while remaining `kubectl exec`'d in the Consul Agent pod.
+
+        ```shell-session
+        consul keyring -use="Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w="
+        ==> Changing primary gossip encryption key...
+        ```
+
+    1. You can ensure that the key has been propagated to all agents by verifying the number of agents that recognize the key over the number of total agents in the datacenter. Listing them provides that information.
+
+        ```shell-session
+        consul keyring -list
+        ==> Gathering installed encryption keys...
+
+        WAN:
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [2/2]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [2/2]
+
+        dc1 (LAN):
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [4/4]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [4/4]
+
+        dc2 (LAN):
+          CL6M+jKj3630CZLXI0IRVeyci1jgIAveiZKvdtTybbA= [4/4]
+          Wa6/XFAnYy0f9iqVH2iiG+yore3CqHSemUy4AIVTa/w= [4/4]
+        ```
+
+1. Update the Kubernetes secrets with the latest gossip encryption key.
+
+    In order for subsequent `helm upgrades` commands to be successfully executed without unexpected behavior, it is important to update the gossip encryption Kubernetes Secret with
+    the value of the new gossip encryption key.
+
+    The name of the secret which stores the value of the gossip encryption key can be found in the Helm values file:
```

Comment:
```suggestion
    The name of the secret that stores the value of the gossip encryption key can be found in the Helm values file:
```
Using 'which' when you mean 'that' is a very common error. 'Which' should be used with commas to set off additional, but unnecessary information, within the phrase. Use 'that' when adding  information critical to the thought conveyed in the sentence. Sorry for being so pedantic.